File: 45/45_jump_game_ii.c
#include <stdio.h>


int jump(int* nums, int numsSize) {
    int maxReachable = 0;
    int maxJumps = 0;
    int current = 0;
    for (int i = 0; i < numsSize - 1; i++) {
        if (i + nums[i] > maxReachable) {
            maxReachable = i + nums[i];
        }

        if (i == current) {
            maxJumps++;
            current = maxReachable;    
        }
    }
    return maxJumps;
}

int main() {
    int nums[] = {2,3,1,1,4};
    printf("%d", jump(nums, 5));
    return 0;
}

File: 45/45_jump_game_ii.py
from typing import List

class Solution:
    def jump(self, nums: List[int]) -> int:
        maxJumps = 0
        maxReachable = 0
        current = 0
        jumps = 0
        for i in range(len(nums) - 1):
            if i + nums[i] > maxReachable:
                maxReachable = i + nums[i]
            if i == current:
                jumps += 1
                current = maxReachable
        return jumps

if __name__ == "__main__":
    print(Solution().jump([2,3,0,1,4]))